lib/view_component/base.rb

```diff
@@ -50,6 +50,17 @@ def config
 
     attr_accessor :__vc_original_view_context
 
+    # Compoents can have a cache key that is used to cache the rendered output.
+    #
+    # @return [String]
+    def cache_key
+      @vc_cache_key = if defined?(__vc_cache_args)
```

Comment:
```suggestion
      return @__vc_cache_key if defined?(@__vc_cache_key)
      @__vc_cache_key ||= if defined?(__vc_cache_args)
```

I think we want to memoize this?